src/helpers/__tests__/insights-test.ts

```diff
@@ -0,0 +1,165 @@
+import insights, {
+  writeDataAttributes,
+  readDataAttributes,
+  hasDataAttributes,
+} from '../insights';
+
+const makeDomElement = (html: string): HTMLElement => {
+  document.body.innerHTML = `<div id="content">${html}</div>`;
+  const content = document.querySelector('#content');
+  if (!content) return document.body;
+  return <HTMLElement>content.firstChild;
+};
+
+describe('insights', () => {
+  test('default behaviour', () => {
+    expect(
+      insights('clickedObjectIDsAfterSearch', {
+        objectIDs: ['3'],
+        eventName: 'Add to Cart',
+      })
+    ).toEqual(
+      `data-insights-method="clickedObjectIDsAfterSearch" data-insights-payload='{"objectIDs":["3"],"eventName":"Add to Cart"}'`
```

Comment:
I found that the snapshot produced are harder to read due to all the quotes. 
Since we care about having the correct quotes here, I though the inline snapshot was confusing.

```js
   expect(
      insights('clickedObjectIDsAfterSearch', {
        objectIDs: ['3'],
        eventName: 'Add to Cart',
      })
    ).toMatchInlineSnapshot(
      `"data-insights-method=\\"clickedObjectIDsAfterSearch\\" data-insights-payload='{\\"objectIDs\\":[\\"3\\"],\\"eventName\\":\\"Add to Cart\\"}'"`
    );
```